test/test_transforms.py

```diff
@@ -1794,6 +1549,292 @@ def test_randomness(fn, trans, config, p):
     assert p_value > 0.0001
 
 
+@pytest.mark.parametrize('data_type, mode', [
+    ('float', 'L'),
+    ('byte', 'L'),
+    ('short', 'I;16'),
+    ('int', 'I'),
+])
+def test_1_channel_tensor_to_pil_image(data_type, mode):
+    to_tensor = transforms.ToTensor()
+
+    if data_type == 'float':
+        img_data = torch.Tensor(1, 4, 4).uniform_()
+        expected_output = img_data.mul(255).int().float().div(255).numpy()
+    elif data_type == 'byte':
+        img_data = torch.ByteTensor(1, 4, 4).random_(0, 255)
+        expected_output = img_data.float().div(255.0).numpy()
+    elif data_type == 'short':
+        img_data = torch.ShortTensor(1, 4, 4).random_()
+        expected_output = img_data.numpy()
+    elif data_type == 'int':
+        img_data = torch.IntTensor(1, 4, 4).random_()
+        expected_output = img_data.numpy()
+    else:
+        assert False
+
+    for transform in [transforms.ToPILImage(), transforms.ToPILImage(mode=mode)]:
+        img = transform(img_data)
+        assert img.mode == mode
+        torch.testing.assert_close(expected_output, to_tensor(img).numpy(), check_stride=False)
+
+
+def test_1_channel_tensor_to_pil_image_error():
+    img_data = torch.Tensor(1, 4, 4).uniform_()
+    # 'F' mode for torch.FloatTensor
+    img_F_mode = transforms.ToPILImage(mode='F')(img_data)
+    assert img_F_mode.mode == 'F'
+    torch.testing.assert_close(
+        np.array(Image.fromarray(img_data.squeeze(0).numpy(), mode='F')), np.array(img_F_mode)
+    )
+
+
+@pytest.mark.parametrize('img_data, mode', [
+    (torch.Tensor(4, 4, 1).uniform_().numpy(), 'F'),
+    (torch.ByteTensor(4, 4, 1).random_(0, 255).numpy(), 'L'),
+    (torch.ShortTensor(4, 4, 1).random_().numpy(), 'I;16'),
+    (torch.IntTensor(4, 4, 1).random_().numpy(), 'I'),
+])
+def test_1_channel_ndarray_to_pil_image(img_data, mode):
+    for transform in [transforms.ToPILImage(), transforms.ToPILImage(mode=mode)]:
+        img = transform(img_data)
+        assert img.mode == mode
+        # note: we explicitly convert img's dtype because pytorch doesn't support uint16
+        # and otherwise assert_close wouldn't be able to construct a tensor from the uint16 array
+        torch.testing.assert_close(img_data[:, :, 0], np.asarray(img).astype(img_data.dtype))
+
+
+@pytest.mark.parametrize('mode', [None, 'LA'])
+def test_2_channel_ndarray_to_pil_image(mode):
+    img_data = torch.ByteTensor(4, 4, 2).random_(0, 255).numpy()
```

Comment:
Hi @NicolasHug 

I directly removed the original `verify_img_data` here

https://github.com/pytorch/vision/blob/2a52c2dca73513d0d0c3e2a505aed05e5b9aa792/test/test_transforms.py#L815-L821

Reply:
This looks like the right thing to do!